In [1]:
'''
Author: Gang Hu
Personal Email: huyingguo2007@hotmail.com
Work/School Email: ghu3@ur.rochester.edu
Date: 2023-08-16 23:54:35
LastEditTime: 2023-08-16 23:55:12
LastEditors: Gang Hu
FilePath: \Hackathon-Summer-2023\test.ipynb
'''

# Read and load the data
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import scanpy as sc
import anndata as ad


dirpath = 'test_data\sce.h5ad'

# load the data
adata = sc.read_h5ad(dirpath)
len(adata) # row number


21954

Tutorial on scanpy: https://scanpy.readthedocs.io/en/stable/

In [33]:
!ls test_data

sce.h5ad
sce.rds


In [31]:
!pip install leidenalg

     ---------------------------------------- 0.0/1.6 MB ? eta -:--:--
     -- ------------------------------------- 0.1/1.6 MB 3.3 MB/s eta 0:00:01
     ------------ --------------------------- 0.5/1.6 MB 6.0 MB/s eta 0:00:01
     ----------------------- ---------------- 0.9/1.6 MB 7.4 MB/s eta 0:00:01
     ------------------------------- -------- 1.2/1.6 MB 7.9 MB/s eta 0:00:01
     ---------------------------------------- 1.6/1.6 MB 7.1 MB/s eta 0:00:00
     ---------------------------------------- 0.0/2.9 MB ? eta -:--:--
     ------ --------------------------------- 0.5/2.9 MB 9.8 MB/s eta 0:00:01
     ------------- -------------------------- 1.0/2.9 MB 10.1 MB/s eta 0:00:01
     ------------------- -------------------- 1.4/2.9 MB 10.1 MB/s eta 0:00:01
     -------------------------- ------------- 1.9/2.9 MB 10.3 MB/s eta 0:00:01
     -------------------------------- ------- 2.4/2.9 MB 10.9 MB/s eta 0:00:01
     ------------------------------------- -- 2.7/2.9 MB 9.6 MB/s eta 0:00


[notice] A new release of pip is available: 23.0 -> 23.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [32]:
!pip install louvain

     ---------------------------------------- 0.0/89.5 kB ? eta -:--:--
     ---------------------------------------- 89.5/89.5 kB 2.6 MB/s eta 0:00:00



[notice] A new release of pip is available: 23.0 -> 23.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [2]:
print(adata)

AnnData object with n_obs × n_vars = 21954 × 25269
    uns: 'X_name'


AnnData object with n_obs × n_vars = 21954 × 25269
    uns: 'X_name'

There are 21954 cells and 25269 gene features in the dataset.

In [3]:
adata.var

Empty DataFrame
Columns: []
Index: [A1BG, A1BG-AS1, A1CF, A2M, A2M-AS1, A2ML1, A4GALT, AA415398, AA465934, AA467197, AA474331, AA474408, AA987161, AAAS, AACS, AADAT, AAED1, AAGAB, AAK1, AAMDC, AAMP, AANAT, AAR2, AARD, AARS, AARS2, AARSD1, AASDH, AASDHPPT, AASS, AATF, AATK, AB015752.3, AB041803, ABAT, ABCA1, ABCA10, ABCA12, ABCA2, ABCA3, ABCA5, ABCA6, ABCA7, ABCA8, ABCA9, ABCB1, ABCB10, ABCB4, ABCB5, ABCB6, ABCB7, ABCB8, ABCB9, ABCC1, ABCC10, ABCC11, ABCC2, ABCC3, ABCC4, ABCC5, ABCC6, ABCC8, ABCC9, ABCD1, ABCD2, ABCD3, ABCD4, ABCE1, ABCF1, ABCF2, ABCF3, ABCG1, ABCG2, ABCG4, ABHD1, ABHD10, ABHD11, ABHD12, ABHD12B, ABHD13, ABHD14A, ABHD14B, ABHD15, ABHD16A, ABHD16B, ABHD17A, ABHD17B, ABHD17C, ABHD2, ABHD3, ABHD4, ABHD5, ABHD6, ABHD8, ABI1, ABI2, ABI3, ABI3BP, ABL1, ABL2, ...]

[25269 rows x 0 columns]

In [4]:
adata.obs

Empty DataFrame
Columns: []
Index: [AGAGCGAAGATCTGCT, ACATCAGTCTGACCTC, TTCTCCTGTCCTGCTT, GAAACTCTCATGTAGC, CAGTCCTGTCTAGCCG, CATGCCTCACACCGCA, AGGCCGTAGGCCCGTT, AAGTCTGAGCGTCTAT, CAGTAACCAGGAACGT, GTCTCGTTCGTACCGG, CCACGGAAGCACACAG, CGCTATCAGTGAAGTT, CGGAGCTCACATTTCT, AGATTGCAGCTGATAA, AGATTGCGTTTGGGCC, AGCGTATTCGACCAGC, GTGTTAGTCGTCGTTC, ACGGGCTCAGGTCTCG, TACTCATAGTACGTTC, CAACCTCGTACGCACC, CGATGTATCTCACATT, GATCGTAGTAAATGTG, CATGACATCTATCGCC, CATATGGCAACTGCGC, ACCGTAAGTCTAGCGC, TGGTTCCAGTCCTCCT, AGCAGCCAGTGCCATT, GGCGTGTTCAGGCAAG, CACAAACAGTGCGATG, ATGGGAGCAGATCGGA, ATCACGATCACTATTC, CTTCTCTAGCCCTAAT, AACTGGTTCGGTTCGG, GACAGAGAGCCCGAAA, ACGATGTAGCGATGAC, CCGTGGACATCTACGA, GGAACTTCATAGGATA, TCGGGACCAAAGAATC, TACACGATCGCTTAGA, ATGTGTGTCCAAAGTC, GCTGCGATCTTCGGTC, GCAATCACAGCTCGAC, GCAGTTACACATGGGA, ATCATCTTCTGCTGCT, GCTGGGTAGACTCGGA, TTCTACACACCGTTGG, CTCGTACCAATGGAGC, CGAGAAGCATGCGCAC, TGAGGGACAGCCTGTG, GGACAGATCCCAAGAT, CATTCGCGTACCAGTT, GTTTCTAAGTCGATAA, CCCATACTCGGTGTTA, ACTTTCAAGATGAGAG, CGCTATCTCCTTTACA, GGGTCTGCAGACTCGC, AAAGATGTCCGCGCAA, ACCCACTGTAAGTGTA, GTCAAGTCAATCCAAC, GGGATGAGTCGAACAG, CTGCGGAGTGTGACGA, TCGTAGAGTAAATACG, TATCTCACACGAGGTA, CGAATGTCAAGGCTCC, CACCTTGCATCACGTA, CACATTTGTATGAAAC, GTCCTCATCCTTGGTC, GGACATTCATTAGCCA, GATCTAGGTCAATACC, TGACAACGTAAGGGCT, TAGTGGTAGGCCCGTT, TGACTAGCACAGGCCT, CCCAATCCAATCGGTT, AGAATAGAGTGACTCT, ACACCAATCCGTCAAA, GTAACGTAGAGTACCG, GCGCGATCATGGAATA, AAACGGGGTGCAACTT, TGAAAGACAAGGTGTG, TGCTGCTAGGGTTTCT, GCTGCGACATGCCTAA, CGTTCTGCAATGGAGC, CCACGGACAAGTAATG, TTATGCTTCTCGGACG, GGAATAATCCATTCTA, TTGAACGCAATTGCTG, TGGCTGGCAGGACGTA, TGGTTCCTCGCCTGAG, GGATGTTCAAGAAGAG, GAACGGATCACAGGCC, CTGCGGATCTTGTACT, TTGACTTGTAACGTTC, TTTACTGGTTGATTGC, AACTCAGGTAGGACAC, AGGCCGTTCACATACG, GGACGTCGTACACCGC, AGCGGTCGTACGACCC, CTCGTCACAGGACGTA, GAGGTGAGTAATCACC, GGGCACTGTCTTTCAT, ...]

[21954 rows x 0 columns]

In [5]:
adata.X


<21954x25269 sparse matrix of type '<class 'numpy.float64'>'
	with 11039225 stored elements in Compressed Sparse Column format>

In [6]:
print(adata.uns['X_name'])


counts


In [7]:
print(adata.uns.keys())


dict_keys(['X_name'])


In [10]:
print(adata.shape)

(21954, 25269)


In [11]:
print(adata.filename)

None


In [42]:
print(adata.var[:20])

Empty DataFrame
Columns: []
Index: [A1BG, A1BG-AS1, A1CF, A2M, A2M-AS1, A2ML1, A4GALT, AA415398, AA465934, AA467197, AA474331, AA474408, AA987161, AAAS, AACS, AADAT, AAED1, AAGAB, AAK1, AAMDC]


In [27]:
df = pd.DataFrame(adata.X)
print(df.head())


                                                   0
0    (0, 14)\t1.0\n  (0, 16)\t2.0\n  (0, 17)\t2.0...
1    (0, 4)\t1.0\n  (0, 13)\t1.0\n  (0, 16)\t5.0\...
2    (0, 4)\t1.0\n  (0, 13)\t3.0\n  (0, 16)\t6.0\...
3    (0, 13)\t1.0\n  (0, 15)\t1.0\n  (0, 17)\t1.0...
4    (0, 13)\t3.0\n  (0, 15)\t1.0\n  (0, 16)\t3.0...


In [28]:
print(adata.obs.head())


Empty DataFrame
Columns: []
Index: [AGAGCGAAGATCTGCT, ACATCAGTCTGACCTC, TTCTCCTGTCCTGCTT, GAAACTCTCATGTAGC, CAGTCCTGTCTAGCCG]


In [29]:
print(adata.var.head())


Empty DataFrame
Columns: []
Index: [A1BG, A1BG-AS1, A1CF, A2M, A2M-AS1]
